packages/aws-cdk-lib/aws-codepipeline-actions/lib/codecommit/source-action.ts

```diff
@@ -54,6 +55,27 @@ export interface CodeCommitSourceVariables {
   readonly commitMessage: string;
 }
 
+/**
+ * Represents a custom event rule in AWS CodePipeline Actions.
+ *
+ * This interface defines the structure for specifying a custom event rule
+ * in the AWS CodePipeline Actions module. The event rule is defined by an
+ * event pattern and a target.
+ *
+ * @see https://docs.aws.amazon.com/codecommit/latest/userguide/monitoring-events.html
+ * @see https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events_targets-readme.html
+ */
+export interface ICustomEventRule {
+  /** event pattern when this rule should be triggered */
+  eventPattern: EventPattern;
+  /** Target e.g. Lambda when event pattern is fulfilled */
+  target: IRuleTarget
+  /** Rulename */
+  ruleName?: string;
+  /** Description */
+  description?: string;
```

Comment:
```suggestion
  /** 
   * Event pattern when this rule should be triggered 
   */
  eventPattern: EventPattern;
  
  /** 
   * Target e.g. Lambda when event pattern is fulfilled 
   */
  target: IRuleTarget
  
  /** 
   * Rulename 
   */
  ruleName?: string;
  
  /** 
   * Description 
   */
  description?: string;
```

Reply:
Minor formatting change, you may have to run the linter afterwards. Also is there any reason we don't want these fields to be `readonly`?

Reply:
Hmm I thought I did. You're right. It makes sense to make them `readonly`. Will change that.